osquery/filesystem/tests/fileops_tests.cpp

```diff
@@ -348,6 +348,127 @@ TEST_F(FileOpsTests, test_large_read_write) {
   }
 }
 
+TEST_F(FileOpsTests, test_chmod_no_exec) {
+  TempFile tmp_file;
+  std::string path = tmp_file.path();
+
+  {
+    PlatformFile fd(path, PF_CREATE_ALWAYS | PF_WRITE);
+    EXPECT_TRUE(fd.isValid());
+    EXPECT_EQ(4, fd.write("TEST", 4));
+  }
+
+  EXPECT_TRUE(platformChmod(path, S_IRUSR | S_IWUSR | S_IROTH | S_IWOTH));
+
+  {
+    PlatformFile fd(path, PF_OPEN_EXISTING | PF_READ);
+    EXPECT_TRUE(fd.isValid());
+
+    auto status = fd.isExecutable();
+    EXPECT_TRUE(!status.ok());
+    EXPECT_EQ(1, status.getCode());
+  }
+}
+
+TEST_F(FileOpsTests, test_chmod_no_read) {
+  TempFile tmp_file;
+  std::string path = tmp_file.path();
+
+  {
+    PlatformFile fd(path, PF_CREATE_ALWAYS | PF_WRITE);
+    EXPECT_TRUE(fd.isValid());
+    EXPECT_EQ(4, fd.write("TEST", 4));
+  }
+
+  EXPECT_TRUE(platformChmod(path, S_IWUSR | S_IWOTH));
+
+  {
+    PlatformFile fd(path, PF_OPEN_EXISTING | PF_READ);
+    EXPECT_FALSE(fd.isValid());
+  }
+
+  {
+    PlatformFile fd(path, PF_OPEN_EXISTING | PF_WRITE);
+    EXPECT_TRUE(fd.isValid());
+  }
+}
+
+TEST_F(FileOpsTests, test_chmod_no_write) {
+  TempFile tmp_file;
+  std::string path = tmp_file.path();
+
+  {
+    PlatformFile fd(path, PF_CREATE_ALWAYS | PF_WRITE);
+    EXPECT_TRUE(fd.isValid());
+    EXPECT_EQ(4, fd.write("TEST", 4));
+  }
+
+  EXPECT_TRUE(platformChmod(path, S_IRUSR | S_IROTH));
+
+  {
+    PlatformFile fd(path, PF_OPEN_EXISTING | PF_READ);
+    EXPECT_TRUE(fd.isValid());
+  }
+
+  {
+    PlatformFile fd(path, PF_OPEN_EXISTING | PF_WRITE);
+    EXPECT_FALSE(fd.isValid());
+  }
+}
+
+#ifdef WIN32
+// Windows-specific test for file "immutability"
+TEST_F(FileOpsTests, test_immutable) {
```

Comment:
Doesn't the `osquery::isImmutable` method exist in POSIX too? Why aren't we testing both platforms?